ngraph/core/src/op/ctc_greedy_decoder.cpp

```diff
@@ -61,3 +73,69 @@ shared_ptr<Node> op::CTCGreedyDecoder::clone_with_new_inputs(const OutputVector&
     check_new_args_count(this, new_args);
     return make_shared<CTCGreedyDecoder>(new_args.at(0), new_args.at(1), m_ctc_merge_repeated);
 }
+namespace
+{
+    bool evaluate_ctc_greedy_decoder(const HostTensorVector& inputs,
+                                     const HostTensorPtr& out,
+                                     bool ctc_merge_repeated)
+    {
+        auto data = inputs[0];
+        auto sequence_masks = inputs[1];
+        switch (data->get_element_type())
+        {
+        case element::Type_t::bf16:
+        {
+            runtime::reference::ctc_greedy_decoder<bfloat16>(
+                data->get_data_ptr<bfloat16>(),
+                sequence_masks->get_data_ptr<bfloat16>(),
+                out->get_data_ptr<bfloat16>(),
+                data->get_shape(),
+                sequence_masks->get_shape(),
+                out->get_shape(),
+                ctc_merge_repeated);
+            break;
+        }
+        case element::Type_t::f16:
+        {
+            runtime::reference::ctc_greedy_decoder<float16>(data->get_data_ptr<float16>(),
+                                                            sequence_masks->get_data_ptr<float16>(),
+                                                            out->get_data_ptr<float16>(),
+                                                            data->get_shape(),
+                                                            sequence_masks->get_shape(),
+                                                            out->get_shape(),
+                                                            ctc_merge_repeated);
+            break;
+        }
+        case element::Type_t::f32:
+        {
+            runtime::reference::ctc_greedy_decoder<float>(data->get_data_ptr<float>(),
+                                                          sequence_masks->get_data_ptr<float>(),
+                                                          out->get_data_ptr<float>(),
+                                                          data->get_shape(),
+                                                          sequence_masks->get_shape(),
+                                                          out->get_shape(),
+                                                          ctc_merge_repeated);
+            break;
+        }
+        case element::Type_t::f64:
```

Comment:
As in reference implementation for ROIAlign we probably need to remove this f64 of this reference implementation too

Reply:
Yes, its weird- I`ve removed it before opening the PR and it took a while to update